openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/LineageRepository.java

```diff
@@ -81,13 +93,55 @@ public void addLineage(AddLineage addLineage) {
 
     // Finally, add lineage relationship
     dao.relationshipDAO()
-        .insert(
-            from.getId(),
-            to.getId(),
-            from.getType(),
-            to.getType(),
-            Relationship.UPSTREAM.ordinal(),
-            detailsJson);
+        .insert(from.getId(), to.getId(), from.getType(), to.getType(), Relationship.UPSTREAM.ordinal(), detailsJson);
+    addLineageToSearch(from, to, addLineage.getEdge().getLineageDetails());
+  }
+
+  private void addLineageToSearch(EntityReference fromEntity, EntityReference toEntity, LineageDetails lineageDetails) {
+    IndexMapping sourceIndexMapping = Entity.getSearchRepository().getIndexMapping(fromEntity.getType());
+    String sourceIndexName = sourceIndexMapping.getIndexName();
+    IndexMapping destinationIndexMapping = Entity.getSearchRepository().getIndexMapping(toEntity.getType());
+    String destinationIndexName = destinationIndexMapping.getIndexName();
+    HashMap<String, Object> relationshipDetails = new HashMap<>();
+    Pair<String, String> from = new ImmutablePair<>("_id", fromEntity.getId().toString());
+    Pair<String, String> to = new ImmutablePair<>("_id", toEntity.getId().toString());
+    processLineageData(fromEntity, toEntity, lineageDetails, relationshipDetails);
+    searchClient.updateLineage(sourceIndexName, from, relationshipDetails);
+    searchClient.updateLineage(destinationIndexName, to, relationshipDetails);
+  }
+
+  private void processLineageData(
+      EntityReference fromEntity,
+      EntityReference toEntity,
+      LineageDetails lineageDetails,
+      HashMap<String, Object> relationshipDetails) {
+    HashMap<String, Object> fromDetails = new HashMap<>();
+    HashMap<String, Object> toDetails = new HashMap<>();
+    fromDetails.put("id", fromEntity.getId().toString());
+    fromDetails.put("type", fromEntity.getType());
+    fromDetails.put("fqn", fromEntity.getFullyQualifiedName());
+    toDetails.put("id", toEntity.getId().toString());
+    toDetails.put("type", toEntity.getType());
+    toDetails.put("fqn", toEntity.getFullyQualifiedName());
+    relationshipDetails.put("doc_id", fromEntity.getId().toString() + "-" + toEntity.getId().toString());
+    relationshipDetails.put("fromEntity", fromDetails);
+    relationshipDetails.put("toEntity", toDetails);
+    if (lineageDetails != null) {
+      relationshipDetails.put(
+          "pipeline",
```

Comment:
this could be pipeline or stored procedure as an edge cc @pmbrull

Reply:
@07Himank need to address this, it can be pipeline or stored procedure here

Reply:
@harshach just key name is pipeline but it works for both pipeline and stored procedure .. do you want to me change the key name?